types/oidcsettings.pp

```diff
@@ -1,113 +1,148 @@
+    Optional['StripCookies']                               => String,
+    Optional['StateMaxNumberOfCookies']                    => Pattern[/^[0-9]+(\s?(false|true))?$/],
+    Optional['SessionInactivityTimeout']                   => Integer,
+    Optional['SessionMaxDuration']                         => Integer,
+    Optional['SessionType']                                => Pattern[/^(server-cache(:persistent)?|client-cookie(:persistent|:store_id_token|:persistent:store_id_token)?)$/],
+    Optional['SessionCacheFallbackToCookie']               => Enum['On', 'Off'],
+    Optional['CacheType']                                  => Enum['shm', 'memcache', 'file', 'redis'],
+    Optional['CacheDir']                                   => String,
+    Optional['CacheEncrypt']                               => Enum['On', 'Off'],
+    Optional['CacheShmMax']                                => Integer,
+    Optional['CacheShmEntrySizeMax']                       => Integer,
+    Optional['CacheFileCleanInterval']                     => Integer,
+    Optional['MemCacheServers']                            => String,
+    Optional['MemCacheConnectionsHMax']                    => Integer,
+    Optional['MemCacheConnectionsMin']                     => Integer,
+    Optional['MemCacheConnectionsSMax']                    => Integer,
+    Optional['MemCacheConnectionsTTL']                     => Integer,
+    Optional['RedisCacheServer']                           => String,
+    Optional['RedisCachePassword']                         => String,
+    Optional['RedisCacheConnectTimeout']                   => Pattern[/^[0-9]+\s?[0-9]*$/],
+    Optional['RedisCacheDatabase']                         => Integer,
+    Optional['RedisCacheTimeout']                          => Integer,
+    Optional['RedisCacheUsername']                         => String,
+    Optional['DiscoverURL']                                => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['HTMLErrorTemplate']                          => String,
+    Optional['DefaultURL']                                 => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['PathScope']                                  => Pattern[/^\"?[A-Za-z0-9\-\._\s]+\"?$/],
+    Optional['PathAuthRequestParams']                      => Pattern[/^[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+(&[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+)*$/],
+    Optional['IDTokenIatSlack']                            => Integer,
+    Optional['ClaimPrefix']                                => String,
+    Optional['ClaimDelimiter']                             => Pattern[/^.$/],
+    Optional['RemoteUserClaim']                            => String,
+    Optional['PassIDTokenAs']                              => Pattern[/^((claims|payload|serialized)\s?)+$/],
+    Optional['PassUserInfoAs']                             => Pattern[/^((claims|json(:([A-Za-z0-9\-\._])+)?|(signed_)?jwt(:([A-Za-z0-9\-\._])+)?)\s?)+$/],
```

Comment:
That looks fragile. Maybe matching something simpler like `:.+` for the ":<name>" would allow some (extra) invalid inputs but would produce a more readable error messages when the provided value is not accepted?

A reliable way of handling this would have been to have complex data type like an `Array[Variant[Enum['claims', 'json', 'jwt', 'signed_jwt'], Struct[{ json => String[1], jwt => String[1], signed_jwt => String[1] }]]]` and some serialization code but that seems to late to introduce this, or at least this should happen only in a major version of the module so not something for this PR.

Reply:
This is anyway beyond the scope of this PR so I will resolve this conversation.